olive/model.py

```diff
@@ -444,3 +467,78 @@ def huggingface_model_loader(model_loader):
         raise ValueError("model_loader must be a callable or a string defined in transformers")
 
     return model_loader.from_pretrained
+
+
+class DistributedOnnxModel(OliveModel):
+    EXECUTION_PROVIDERS = {
+        "cpu": ["CPUExecutionProvider"],
+        "gpu": ["CUDAExecutionProvider", "CPUExecutionProvider"],
+    }
+
+    def __init__(
+        self,
+        model_filepaths: List[str],
+        name: Optional[str] = None,
+        version: Optional[int] = None,
+        inference_settings: Optional[dict] = None,
+    ):
+        super().__init__(
+            framework=Framework.ONNX,
+            model_path=None,
+            name=name,
+            version=version,
+            is_file=False,
+            is_aml_model=False,
+        )
+        self.model_filepaths = model_filepaths
+        self.inference_settings = inference_settings
+
+    @property
+    def ranks(self):
+        return len(self.model_filepaths)
+
+    def load_model(self, rank: int) -> ONNXModel:
+        return ONNXModel(self.model_filepaths[rank], inference_settings=self.inference_settings)
+
+    def prepare_session(self, inference_settings: Optional[Dict[str, Any]] = None,
+                        device: Device = Device.GPU, rank: int = 0):
+        import onnxruntime as ort
+
+        ort_inference_settings = inference_settings or self.inference_settings
+        ort_session_options = _to_ort_session_options(ort_inference_settings)
```

Comment:
How about make DistributedOnnxModel inherit ONNXModel but no OliveModel?

Reply:
That doesn't work well because model_path arg becomes ambiguous. Which model with the collection is the ONNXModel.model_path referring to.  Introduced ONNModelBase that works as a base class for both ONNXModel & DistributedOnnxModel.